github-issue-opener/iac/main.tf

```diff
@@ -41,7 +41,7 @@ resource "google_secret_manager_secret_iam_member" "grant-secret-access" {
   member    = "serviceAccount:${google_service_account.gh-iss-opener.email}"
 }
 
-resource "ko_image" "image" {
+resource "ko_build" "image" {
```

Comment:
I know we've had some "fun" with tf-ko and state updates in the past (I'm thinking of the go module moving I think), so it'd be good to double check that this won't make things angry.

Reply:
I tested this with a locally built TF provider. `terraform plan` gave me this:

```
  # ko_build.example will be created
  + resource "ko_build" "example" {
      + base_image  = "cgr.dev/chainguard/static"
      + id          = (known after apply)
      + image_ref   = (known after apply)
      + importpath  = "github.com/ko-build/terraform-provider-ko/cmd/test"
      + sbom        = "spdx"
      + working_dir = "."
    }

  # ko_image.example will be destroyed
  # (because ko_image.example is not in configuration)
  - resource "ko_image" "example" {
      - base_image  = "gcr.io/distroless/static:nonroot" -> null
      - id          = "gcr.io/jason-chainguard/github.com/chainguard-dev/terraform-provider-ko/cmd/test@sha256:39550436a07ee368ae0f8c18644224634a87dfa9dd35dfdb89cfc694df138cae" -> null
      - image_ref   = "gcr.io/jason-chainguard/github.com/chainguard-dev/terraform-provider-ko/cmd/test@sha256:39550436a07ee368ae0f8c18644224634a87dfa9dd35dfdb89cfc694df138cae" -> null
      - importpath  = "github.com/chainguard-dev/terraform-provider-ko/cmd/test" -> null
      - platforms   = [
          - "linux/amd64",
        ] -> null
      - sbom        = "spdx" -> null
      - working_dir = "." -> null
    }
```

This is the expected behavior for this kind of change AFAICT. If you upgrade to the latest TF provider (when it's released) but don't change `ko_image` -> `ko_build` you'll get an error applying because its `.image_ref` will be `""`, and you'll find that it's because the resource is deprecated in favor of `ko_build`.

We can't just remove the `ko_image` resource, and we can't make it an error to read or delete it, or else you can't remove it from your state.